Lib/asyncore.py

```diff
@@ -127,7 +127,8 @@ def poll(timeout=0.0, map=None):
         map = socket_map
     if map:
         r = []; w = []; e = []
-        for fd, obj in list(map.items()):
+        map_copy = dict(map)
```

Comment:
Can you explain why this is needed? asyncore is not thread safe, and cannot be used by different threads, so the map cannot change while this code is iterating on the map.